### PR TITLE
Embed refresh payload inside a refresh token

### PR DESCRIPTION
**Pull request by lawliet89**

### CODE REVIEW

File: src/routes.rs [hunk 1]
```diff
@@ -154,6 +163,12 @@ mod tests {
             signature_algorithm: Some(jwt::jwa::SignatureAlgorithm::HS512),
             secret: Secret::String("secret".to_string()),
             expiry_duration: Duration::from_secs(120),
+            refresh_token: Some(RefreshTokenConfiguration {
+                                    cek_algorithm: jwt::jwa::KeyManagementAlgorithm::A256GCMKW,
+                                    enc_algorithm: jwt::jwa::ContentEncryptionAlgorithm::A256GCM,
+                                    key: jwt::jwk::JWK::new_octect_key(&[0; 256 / 8], Default::default()),
+                                    expiry_duration: Duration::from_secs(80640),
```
Comment (gyng) [inline src/routes.rs]: 86400 or configured?

File: src/token.rs [hunk 2]
```diff
@@ -546,31 +819,99 @@ mod tests {
...
+                          jwt::jwa::KeyManagementAlgorithm::A256GCMKW,
+                          jwt::jwa::ContentEncryptionAlgorithm::A256GCM,
+                          UTC::now())
+                .unwrap()
+    }
+
+    fn make_token(refresh_token: bool) -> Token<TestClaims> {
+        let refresh_token = if refresh_token {
+            Some(make_refresh_token())
+        } else {
```
Comment (gyng) [inline src/token.rs:844]: .or(None)

File: src/token.rs [hunk 3]
```diff
@@ -546,31 +819,99 @@ mod tests {
...
+            refresh_token: refresh_token,
+        }
     }
 
     #[test]
-    fn encoding_and_decoding_round_trip() {
-        let token = make_token();
-        let token = not_err!(token.encode(&jwt::jws::Secret::bytes_from_str("secret")));
+    fn refresh_token_encryption_round_trip() {
+        let key = jwt::jwk::JWK::new_octect_key(&[0; 256 / 8], Default::default());
```
Comment (gyng) [inline src/token.rs:862]: Bounds right?

**Review by gyng — APPROVED** [review]

File: src/routes.rs (hunk 1 above)
Comment (lawliet89) [reply src/routes.rs]: This is a test fixture. Configurable. But yeah it's a typo.

File: src/token.rs (hunk 3 above)
Comment (lawliet89) [reply src/token.rs:862]: Key is 256 bits. Easier (for me) to express it in bits than bytes.

**Pull request merged**